### PR TITLE
[FEAT] Allow passing on_error="null" to ignore decoding errors in image decode

### PR DESCRIPTION
**Pull request by jaychia**
Allows for loose decoding of images, when we expect some bytes to be bad images

API:
```
df.with_column("images", df["bytes"].image.decode(on_error="null"))
```

### CODE REVIEW

File: tests/series/test_image.py [hunk 1]
```diff
@@ -709,3 +709,12 @@ def test_bad_cast_fixed_shape_image():
...
     with pytest.raises(ValueError, match="Expected Numpy array to be of type: UInt8"):
         s.cast(target_dtype)
+
+
+def test_on_error_image_decode():
+    s = Series.from_pylist([b"not an image"])
+
+    with pytest.raises(ValueError, match="Decoding image from bytes failed"):
+        s.image.decode(on_error="raise")
+
```
Comment (colin-ho) [inline tests/series/test_image.py:719]: add test for parameter input that isn't `raise` or `null` ?

**Review by colin-ho — APPROVED** [review]
Just left a couple comments/questions!

File: daft/expressions/expressions.py [hunk 2]
```diff
@@ -1076,16 +1076,27 @@ def __repr__(self) -> str:
 class ExpressionImageNamespace(ExpressionNamespace):
     """Expression operations for image columns."""
 
-    def decode(self) -> Expression:
+    def decode(self, on_error: Literal["raise"] | Literal["null"] = "raise") -> Expression:
```
Comment (clarkzinzow) [inline daft/expressions/expressions.py:1079]: If we expect this to always have boolean semantics, a better user-facing API might be `raise_on_error: bool` or the like.

**Review by clarkzinzow — APPROVED** [review]
LGTM overall!

File: daft/expressions/expressions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaychia) [reply daft/expressions/expressions.py:1079]: I was following the semantics of our current `.url.download(on_error: "null" | "raise")`, so that our APIs could be more consistent.

Comment (jaychia) [reply daft/expressions/expressions.py:1079]: I was following the semantics of our current `.url.download(on_error: "null" | "raise")`, so that our APIs could be more consistent.
